src/algebra/algebra/subalgebra.lean

```diff
@@ -807,6 +807,13 @@ subtype.ext rfl
 @[simp] lemma coe_inclusion {S T : subalgebra R A} (h : S ≤ T) (s : S) :
   (inclusion h s : A) = s := rfl
 
+lemma val_injective {R A : Type*} [comm_semiring R] [semiring A] [algebra R A]
```

Comment:
I don't think we should have this lemma, we tend not to repeat results for bundled homes. Probably you can just use `subtype.coe_injective` where you were planning to use this lemma